build: adds DeBankNetWorth to intercom

matthew-garrett opened the pull request.

jeffywu commented on packages/shared/wallet/.eslintrc.json:
```diff
@@ -18,7 +18,9 @@
     {
       "files": ["*.ts", "*.tsx"],
-      "rules": {}
+      "rules": {
```
Instead of turning this off, you should add the words to the word list by installing the cspell extension

jeffywu commented on packages/shared/wallet/src/hooks/use-intercom-update.tsx:
```diff
@@ -1,16 +1,63 @@
+      setNetWorth(Math.trunc(netWorth));
+      setInLocalStorage('userSettings', {
+        ...userSettings,
+        debankAddress: selectedAccount,
```
Why do we need to set either of these in local storage? The net worth will change over time.

matthew-garrett replied:
The debankAddress is in local storage to prevent excess api calls because we have to pay for them. The debankNetworth is in local storage so that if someone disconnects their wallet and connects again and the connection is recorded in intercom the debankNetworth is still sent and not zero

jeffywu commented on packages/shared/wallet/src/hooks/use-intercom-update.tsx:
```diff
@@ -1,16 +1,63 @@
   const isReadOnlyAddress = wallet?.isReadOnlyAddress;
   const selectedAccount = wallet?.selectedAddress;
 
+  if (
```
This should also be inside a useEffect, I don't think its good to run promises inside the render cycle like this

matthew-garrett replied:
I kept this out of the useEffect because I think it will likely cause more renders being inside than in. I'll go ahead and move it in if you want

jeffywu commented on packages/shared/wallet/src/hooks/use-intercom-update.tsx:
```diff
@@ -1,16 +1,63 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+
+const DeBankAPIKey = process.env['NX_DE_BANK_API_KEY'] as string | undefined;
```
it should be "NX_DEBANK_API_KEY", debank is one word

matthew-garrett replied:
Typescript freaked out about this being DEBANK. I had it that originally. When I separated the words it was fine

matthew-garrett committed:
build: adds DeBankNetWorth to intercom

build: updates error

build: updates fetchDebank

build: updates fetchDebank

build: removes debounce after further testing

jeffywu commented on packages/shared/wallet/src/hooks/use-intercom-update.tsx:
```diff
@@ -1,22 +1,101 @@
+      setIsFetching(true);
+      try {
+        const netWorth = await fetchDeBankNetWorth(address);
+        const currentTimestamp = new Date().toISOString();
```
Use `getNowSeconds() instead.

jeffywu commented on packages/shared/wallet/src/hooks/use-intercom-update.tsx:
```diff
@@ -1,22 +1,101 @@
+  const weeklyCheck = (() => {
+    if (userSettings.debankTimestamp) {
+      const lastUpdateDate = new Date(userSettings.debankTimestamp);
+      const currentDate = new Date();
```
Use `getNowSeconds()` here as well

matthew-garrett committed:
build: addresses comments

jeffywu commented on packages/shared/wallet/.cspell.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "words": ["debank", "leaderboard", "Avenir"]
```
add these words as lines to the notional-words.txt

matthew-garrett committed:
build: updates time stamps

matthew-garrett committed:
build: more timestamp changes
